versions/schematic-2.md

```diff
@@ -0,0 +1,196 @@
+`String` | `NBT String` | No conversion
+`Object` | `NBT Compound` | No conversion, each object should serialize to supported types within a compound
+
+
+## Specification
+
+### Format
+
+The structure described by this specification is persisted to disk using the [Named Binary Tag](http://minecraft.gamepedia.com/NBT_format) (NBT) format. Before writing to disk the NBT data must be compressed using the [GZip](https://www.gnu.org/software/gzip/) data compression algorithm. The highly recommended file extension for files using this specification is `.schem` (chosen so as to not conflict with the legacy `.schematic` format allowing easy distinction between the two).
+
+All field names in the specification are **case sensitive**.
+
+### Schema
+
+#### <a name="schematicObject"></a>Schematic Object
+
+This is the root object for the specification.
+
+##### Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="schematicVersion"></a>Version | `integer` | **Required.** Specifies the format version being used. It may be used to provide validation and auto-conversion from older versions. The current version is `2`.
+<a name="schematicDataVersion"></a>Data Version | `integer` | **Required.** Specifies the data version of Minecraft that was used to create the schematic. This is to allow for block and entity data to be validated and auto-conversion from older versions. This is dependent on the Minecraft version, eg. Minecraft 1.12.2's data version is [1343](https://minecraft.gamepedia.com/1.12.2).
+<a name="schematicMetadata"></a>Metadata | [Metadata Object](#metadataObject) | Provides optional metadata about the schematic.
+<a name="schematicWidth"></a>Width | `unsigned short` | **Required.** Specifies the width (the size of the area in the X-axis) of the schematic.
+<a name="schematicHeight"></a>Height | `unsigned short` | **Required.** Specifies the height (the size of the area in the Y-axis) of the schematic.
+<a name="schematicLength"></a>Length | `unsigned short` | **Required.** Specifies the length (the size of the area in the Z-axis) of the schematic.
+<a name="schematicOffset"></a>Offset | `integer`[3] | Specifies the relative offset of the schematic. This value is relative to the most minimal point in the schematics area. The default value if not provided is `[0, 0, 0]`. This may be used when incorporating the area of blocks defined by this schematic to a larger area to place the blocks relative to a selected point.
+<a name="schematicPaletteMax"></a>PaletteMax | `integer` | Specifies the size of the block palette in number of bytes needed for the maximum palette index. Implementations may use this as a hint for the case that the palette data fits within a datatype smaller than a 32-bit integer that they may allocate a smaller sized array.
```

Comment:
Wording is a bit off here. I'd clarify that it's just the `max index - 1`(technically not size since the spec allows for gaps in the palette?) instead of a number of bytes.